overlord/snapstate/conflict.go

```diff
@@ -123,6 +123,12 @@ func CheckChangeConflictMany(st *state.State, instanceNames []string, ignoreChan
 			continue
 		}
 
+		if task.Kind() == "update-gadget" {
+			// make sure no other changes are running when a change
+			// that updates the gadget snap is in progress
+			return &ChangeConflictError{Message: "gadget update in progress"}
```

Comment:
as we discussed might have unwanted effects in scenarios like:  upddate many of a gadget and of a snap that needs a new default provider for example, because of the Install coming from doPrerequisite.  We either need to make that code self-conflict aware (it might need to become anyway, we might have already bugs related to that), or use a different strategy here given that this is about exactly one task, either by using the blocked predicates or just the state Lock (though keeping that for really very long might have unexpected effects) ?

Reply:
Making the check self aware seems easiest. I need to review how the blocked predicate works in the grand scheme of things. I see we're using it in hookstate and ifacestate, so at least there's some prior use cases to analyze.

Reply:
> Making the check self aware seems easiest.

not sure what you have in mind but easy is not the word I would use :)

If we just pass in ignoreChangeID and use it, we might get effects we don't want,  we do need to detect self conflicts sometimes, it depends.  I fear we might have already some bugs in the area, and this probably needs another round of improvement.

Reply:
do we need this if we have the blocked predicate now?

Reply:
I think we could keep it. This will block new changes from being added (at least ones that we do the check for). Then, the slot for update-gadget task to run alone should come sooner if there are fewer tasks to run. What do you think?

Reply:
but we haven't solved (yet) my concerns around self-conflicts though, it would be mostly an optimisation, I think if we do it given that main quiescence is checked by the blocked predicate we would should do it later as a separate PR

Reply:
Ok, I'll revert the initial commit and keep the blocking predicate only.

Reply:
Dropped the change conflicts checks now.